services/web/server/tests/helpers/utils_docker.py

```diff
@@ -0,0 +1,68 @@
+
+import logging
+import os
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import docker
+import yaml
+from tenacity import after_log, retry, stop_after_attempt, wait_fixed
+
+log = logging.getLogger(__name__)
+
+@retry(
+    wait=wait_fixed(2),
+    stop=stop_after_attempt(10),
+    after=after_log(log, logging.WARN))
+def get_service_published_port(service_name: str) -> str:
+    # WARNING: ENSURE that service name defines a port
+    # NOTE: retries since services can take some time to start
+    client = docker.from_env()
+    services = [x for x in client.services.list() if service_name in x.name]
+    if not services:
+        raise RuntimeError("Cannot find published port for service '%s'. Probably services still not up" % service_name)
```

Comment:
per se, the task does not need to be up/running to get a port. but the service needs to be started. maybe the error message could be adapted.